src/imitation/scripts/analyze.py

```diff
@@ -74,62 +108,174 @@ def gather_tb_directories(
+    else:
+        return f"??exp_command={exp_command}"
+
+
+def _return_summaries(sd: sacred_util.SacredDicts) -> dict:
+    imit_stats = get(sd.run, "result.imit_stats")
+    expert_stats = get(sd.run, "result.expert_stats")
+
+    expert_return_summary = None
+    if expert_stats is not None:
+        expert_return_summary = _make_return_summary(expert_stats)
+
+    imit_return_summary = None
+    if imit_stats is not None:
+        imit_return_summary = _make_return_summary(imit_stats, "monitor_")
+
+    if imit_stats is not None and expert_stats is not None:
+        # Assuming here that `result.imit_stats` and `result.expert_stats` are
+        # formatted correctly.
+        imit_expert_ratio = (
+            imit_stats["monitor_return_mean"] / expert_stats["return_mean"]
+        )
+    else:
+        imit_expert_ratio = None
+
+    return dict(
+        expert_stats=expert_stats,
+        imit_stats=imit_stats,
+        expert_return_summary=expert_return_summary,
+        imit_return_summary=imit_return_summary,
+        imit_expert_ratio=imit_expert_ratio,
+    )
+
+
+sd_to_table_entry_type = Mapping[str, Callable[[sacred_util.SacredDicts], Any]]
+
+# This OrderedDict maps column names to functions that get table entries, given the
+# row's unique SacredDicts object.
+table_entry_fns: sd_to_table_entry_type = collections.OrderedDict(
+    [
+        ("status", lambda sd: get(sd.run, "status")),
+        ("exp_command", _get_exp_command),
+        ("algo", _get_algo_name),
+        ("env_name", lambda sd: get(sd.config, "env_name")),
+        ("n_expert_demos", lambda sd: get(sd.config, "n_expert_demos")),
+        ("run_name", lambda sd: get(sd.run, "experiment.name")),
+        (
+            "expert_return_summary",
+            lambda sd: _return_summaries(sd)["expert_return_summary"],
+        ),
+        (
+            "imit_return_summary",
+            lambda sd: _return_summaries(sd)["imit_return_summary"],
+        ),
+        ("imit_expert_ratio", lambda sd: _return_summaries(sd)["imit_expert_ratio"]),
+    ]
+)
+
+
+# If `verbosity` is at least the length of this tuple, then we use all table_entry_fns
+# as columns of table.
+# Otherwise, use only the subset at index `verbosity`. The subset of columns is
+# still arranged in the same order as in the `table_entry_fns` OrderedDict.
+table_verbosity_mapping: List[Set[str]] = []
+
+# verbosity 0
+table_verbosity_mapping.append(
+    {
+        "algo",
+        "env_name",
+        "expert_return_summary",
+        "imit_return_summary",
+    }
+)
+
+# verbosity 1
+table_verbosity_mapping.append(table_verbosity_mapping[-1] | {"n_expert_demos"})
+
+# verbosity 2
+table_verbosity_mapping.append(
+    table_verbosity_mapping[-1]
+    | {"status", "imit_expert_ratio", "exp_command", "run_name"}
+)
+
+
+def _get_table_entry_fns_subset(table_verbosity: int) -> sd_to_table_entry_type:
+    assert table_verbosity >= 0
+    if table_verbosity >= len(table_entry_fns):
```

Comment:
```suggestion
    if table_verbosity >= len(table_verbosity_mapping):
```
given it's used as an index into `table_verbosity_mapping` on line 217